src/v/cloud_storage/topic_manifest.cc

```diff
@@ -39,8 +39,909 @@
+        "record_key_subject_name_strategy_compat",
+        &tp::record_key_subject_name_strategy_compat,
+        boost::lexical_cast<
+          std::string,
+          pandaproxy::schema_registry::subject_name_strategy>,
+        boost::lexical_cast<
+          pandaproxy::schema_registry::subject_name_strategy,
+          std::string_view>},
+      type_mapping{
+        "record_value_schema_id_validation",
+        &tp::record_value_schema_id_validation,
+        [](bool out) { return out; },
+        [](bool in) { return in; }},
+      type_mapping{
+        "record_value_schema_id_validation_compat",
+        &tp::record_value_schema_id_validation_compat,
+        [](bool out) { return out; },
+        [](bool in) { return in; }},
+      type_mapping{
+        "record_value_subject_name_strategy",
+        &tp::record_value_subject_name_strategy,
+        boost::lexical_cast<
+          std::string,
+          pandaproxy::schema_registry::subject_name_strategy>,
+        boost::lexical_cast<
+          pandaproxy::schema_registry::subject_name_strategy,
+          std::string_view>},
+      type_mapping{
+        "record_value_subject_name_strategy_compat",
+        &tp::record_value_subject_name_strategy_compat,
+        boost::lexical_cast<
+          std::string,
+          pandaproxy::schema_registry::subject_name_strategy>,
+        boost::lexical_cast<
+          pandaproxy::schema_registry::subject_name_strategy,
+          std::string_view>},
+      type_mapping{
+        "initial_retention_local_target_bytes",
+        &tp::initial_retention_local_target_bytes,
+        [](size_t in) { return uint64_t(in); },
+        []<std::integral I>(I in) {
+            if constexpr (std::signed_integral<I>) {
+                if (in < 0) {
+                    // special case: handle -1 values as disabled (this is not
+                    // the encoding performed by this code for tristate, but
+                    // could happen in manual edits)
+                    return tristate<size_t>{disable_tristate};
+                }
+            }
+            return tristate<size_t>{size_t(in)};
+        }},
+      type_mapping{
+        "initial_retention_local_target_ms",
+        &tp::initial_retention_local_target_ms,
+        &std::chrono::milliseconds::count,
+        []<std::integral I>(I in) {
+            if constexpr (std::signed_integral<I>) {
+                if (in < 0) {
+                    // special case: handle -1 values as disabled (this is not
+                    // the encoding performed by this code for tristate, but
+                    // could happen in manual edits)
+                    return tristate<std::chrono::milliseconds>{
+                      disable_tristate};
+                }
+            }
+            return tristate<std::chrono::milliseconds>{
+              std::chrono::milliseconds(in)};
+        }},
+    };
+};
+
+} // namespace
 namespace cloud_storage {
 
+class topic_properties_handler
```

Comment:
I believe this will benefit from some fuzzing.

Reply:
yes good point